File: src/main/java/supercritical/api/unification/ElementExtension.java
package supercritical.api.unification;

import gregtech.api.unification.Element;
import lombok.experimental.UtilityClass;

/// See [MixinElement]
public interface ElementExtension {

    double getHalfLiveSeconds();

    void setHalfLiveSeconds(double halfLifeSeconds);

    @UtilityClass
    class Handler {

        public double getHalfLiveSeconds(Element element) {
            return ((ElementExtension) element).getHalfLiveSeconds();
        }

        public void setHalfLiveSeconds(Element element, double halfLifeSeconds) {
            ((ElementExtension) element).setHalfLiveSeconds(halfLifeSeconds);
        }
    }
}

File: src/main/kotlin/supercritical/api/unification/ElementExtension.kt
package supercritical.api.unification

/**
 * Extends [Element] with a double-precision half-life in seconds.
 * The modern Element class stores half-life as a long; this mixin converts
 * to a double so Supercritical can represent sub-second isotope lifetimes.
 */
interface ElementExtension {
    var halfLifeSeconds: Double

    val halfLifeSecondsLong: Long
        /**
         * Convenience helper for legacy callers that expect a long value.
         */
        get() = this.halfLifeSeconds.toLong()
}
